af_pipeline/af_input/msa.py

```
@@ -0,0 +1,353 @@
+        """
+
+        tar_name = os.path.basename(targz_file).split(".tar.gz")[0]
+        resultdir = os.path.join(os.path.dirname(targz_file), tar_name)
+        if os.path.exists(resultdir) and not overwrite:
+            logger.error(f"Output directory {resultdir} already exists. Skipping extraction.")
+            return
+
+        shutil.rmtree(resultdir, ignore_errors=True)
+
+        with tarfile.open(targz_file, "r:gz") as tar:
+            tar.extractall(path=resultdir)
```

Ruff (0.16.3)

[error] Uses of `tarfile.extractall()`

(S202)